### PR TITLE
Fix the crash issue on the Android platform caused by an invalid context.

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #
https://github.com/cocos/cocos-engine/issues/17910

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: native/cocos/platform/android/adpf_manager.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ bool ADPFManager::initializePowerManager() {
     #endif
 
     JNIEnv *env = cc::JniHelper::getEnv();
-    auto *javaGameActivity = cc::JniHelper::getActivity();
+    auto *javaGameActivity = cc::JniHelper::getContext();
```
Comment (dumganhar) [inline native/cocos/platform/android/adpf_manager.cpp]: Shall we also rename the variable name to `auto *javaContext` ?

**Review by minggo — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by finscn** [comment]
看了下pr内容.  感觉这个bug是底层最初设计就有问题.  该用application context的地方, 错误的使用了Activity.   如果是这样, 那这个bug岂不是已经存在好几年了,  从cocos支持安卓开始就有?  之前没人反馈过吗?

**Pull request merged**